src/components/forms/GSDateField.jsx

```diff
@@ -4,59 +4,34 @@ import moment from "moment";
 import GSFormField from "./GSFormField";
 import { dataTest } from "../../lib/attributes";
 import DateFnsUtils from "@date-io/date-fns";
-import { MuiPickersUtilsProvider, DatePicker } from "@material-ui/pickers";
+import {
+  MuiPickersUtilsProvider,
+  KeyboardDatePicker
+} from "@material-ui/pickers";
 import theme from "../../styles/mui-theme";
 
-export default class GCDateField extends GSFormField {
-  render() {
-    const momentDate = moment(this.props.value);
-    let value = {};
-    let oldDate = null;
-    if (momentDate.isValid()) {
-      const fakeDate = momentDate
-        .add(this.props.utcOffset - moment().utcOffset(), "minutes")
-        .toDate();
-      oldDate = moment(fakeDate).toObject();
-      value = { value: fakeDate };
-    }
-    const { utcOffset } = this.props;
-    const propCopy = {
-      ...this.props
-    };
-    delete propCopy.utcOffset;
-    delete propCopy.value;
-    delete propCopy.type;
-    const dataTest = { "data-test": this.props["data-test"] };
+export default class GSDateField extends GSFormField {
+  state = { open: false };
 
+  render() {
     return (
       <MuiPickersUtilsProvider utils={DateFnsUtils}>
-        <DatePicker
-          {...dataTest}
+        <KeyboardDatePicker
           fullWidth
+          onKeyPress={evt => {
+            this.setState({ open: true });
+            evt.preventDefault();
+          }}
+          onClose={() => this.setState({ open: false })}
+          onOpen={() => this.setState({ open: true })}
+          autoOk
           format="MM/dd/yyyy"
           label={this.floatingLabelText()}
-          value={value.value}
-          onChange={date => {
-            let newDate = moment(date);
-            if (!newDate.isValid()) {
-              this.props.onChange(null);
-            } else {
-              newDate = newDate.toObject();
-              if (oldDate) {
-                newDate.hours = oldDate.hours;
-                newDate.minutes = oldDate.minutes;
-                newDate.seconds = oldDate.seconds;
-              }
-              newDate = moment(newDate).add(
-                moment().utcOffset() - utcOffset,
-                "minutes"
-              );
-              this.props.onChange(newDate.toDate());
-            }
-          }}
           style={{
             marginBottom: theme.spacing(2)
           }}
+          open={this.state.open}
+          {...this.props}
```

Comment:
`{...this.props}` this is an anti pattern. Modern react throws errors when you pass unexpected unexpected props to components. Also this makes making changes on components extra hard as you never know if a change you are making effect some prop you don't know about. If we can get rid of this and pick the props we need to pass in this will not only get rid of errors but also make future development much easier. 

Do you know what props you were trying to pass through?

Reply:
oh wait. I see where you moved this from. I see you're just kicking the can down the road that I also kicked down the road. So not a regression! But it would be nice to remove all of that spreading on props.

Reply:
Thanks for feedback.
I'm passing through 
https://github.com/MoveOnOrg/Spoke/pull/2025/files/6c525d51da1f44b2724a4ae7cdabea47eb4eca04#diff-ce3ce9d2ab22743073212dd164e9ba673418132730b0c9ad686ea4d3eefeeaf3R95
at least which since this is the only date instance we could push it into there, but I thought it made more sense in the first context.
I get the anti-pattern issue -- so if i tweak again, maybe i'll just pass explicit props, though datefield components have so many, it would be more annoying/confusing to 1. list them all explicitly in the date field or 2. only pass the ones i need, but if someone creates a datefield.
I guess we can do this.props and just remove the ones that were for the Form.Field component.